src/iarc7_motion/task_command_handler.py

```diff
@@ -0,0 +1,238 @@
+            self._ground_interaction_client.send_goal(goal, done_cb=self.handle_ground_interaction_done)
+            self._in_passthrough = True
+        else:
+            self._ground_interaction_client.cancel_goal()
+            self._in_passthrough = False
+
+    def _handle_passthrough_command(self, passthrough_command):
+        if self._in_passthrough:
+            msg = OrientationThrottleStamped()
+            msg.header.stamp = rospy.Time.now()
+            msg.data.pitch = passthrough_command.pitch
+            msg.data.roll = passthrough_command.roll
+            msg.data.yaw = passthrough_command.vyaw
+            msg.throttle = passthrough_command.vz
+            self._passthrough_pub.publish(msg)
+        else:
+            rospy.logerr('Task requested a passthrough command when not in passthrough mode')
+
+    def _handle_ground_interaction_command(self, ground_interaction_command):
+        if self._ground_interaction_task_callback is not None:
+            rospy.logerr('Task requested another ground interaction action before the last completed')
+            rospy.logerr('Task Command Handler aborted task')
+            self._task = None
+            self._task_state = task_states.TaskAborted()
+            self._ground_interaction_task_callback = None
+            self._ground_interaction_client.cancel_goal()
+            self._ground_interaction_client.stop_tracking_goal()
+            return
+
+        self._ground_interaction_task_callback = ground_interaction_command.completion_callback
+        # Request ground interaction of llm
+        goal = GroundInteractionGoal(interaction_type=ground_interaction_command.interaction_type)
+        # Sends the goal to the action server.
+        self._ground_interaction_client.send_goal(goal, done_cb=self.handle_ground_interaction_done)
+    
+    # callback for status on ground interaction commands
+    def handle_ground_interaction_done(self, status, result):
+        if self._ground_interaction_task_callback is not None:
+            try:
+                self._ground_interaction_task_callback(status, result)
+            except Exception as e:
+                rospy.logerr('Exception sending result using ground interaction done cb')
+                rospy.logerr('Task Command Handler aborted task')
+                rospy.logerr(str(e))
+                rospy.logerr(traceback.format_exc())
+                self._task = None
+                self._task_state = task_states.TaskAborted()
+            self._ground_interaction_task_callback = None
+        else:
+            rospy.logerr('Ground interaction done callback received with no task callback available')
+
+    """
+    Sends twist (x, y, z, and angular velocities) to LLM
+
+    Args:
+        twist: TwistStampedArray (array of twists) or just a single twist stamped.
+    """
+    def _publish_twist(self, twist):
+        if type(twist) is TwistStampedArray:
+            self._last_twist = twist.twists[-1]
```

Comment:
We need to cache the last twist being executed.... This could really screw up the acceleration calculations in an emergency stop.

Reply:
right, thats what we are doing here, we are saving off the last one in the list (which is the most future one).